pkg/storage/chunk/client/alibaba/oss_object_client.go

```diff
@@ -52,11 +54,17 @@ func (cfg *OssConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
 	f.StringVar(&cfg.Endpoint, prefix+"oss.endpoint", "", "oss Endpoint to connect to.")
 	f.StringVar(&cfg.AccessKeyID, prefix+"oss.access-key-id", "", "alibabacloud Access Key ID")
 	f.StringVar(&cfg.SecretAccessKey, prefix+"oss.secret-access-key", "", "alibabacloud Secret Access Key")
+	f.Int64Var(&cfg.ConnectionTimeoutSec, prefix+"oss.conn-timeout-sec", 5, "Connection timeout in seconds")
+	f.Int64Var(&cfg.ReadWriteTimeoutSec, prefix+"oss.read-write-timeout-sec", 5, "Read/Write timeout in seconds")
 }
 
 // NewOssObjectClient makes a new chunk.Client that writes chunks to OSS.
 func NewOssObjectClient(_ context.Context, cfg OssConfig) (client.ObjectClient, error) {
-	client, err := oss.New(cfg.Endpoint, cfg.AccessKeyID, cfg.SecretAccessKey)
+	var options []oss.ClientOption
+	if cfg.ConnectionTimeoutSec > 0 || cfg.ReadWriteTimeoutSec > 0 {
```

Comment:
looking at the code, setting `ConnectionTimeoutSec` to 0 means no timeout on connection. but setting `ReadWriteTimeoutSec` to 0 would fail the reads and writes immediately i think 

should we pass these values as is and have a validation on `ReadWriteTimeoutSec` to make sure it is greater than 0?

Reply:
https://github.com/grafana/loki/blob/f2da621f7255c678774fb0e99c0b7fb68f6a6a69/vendor/github.com/aliyun/aliyun-oss-go-sdk/oss/transport_1_7.go#L20
https://github.com/grafana/loki/blob/f2da621f7255c678774fb0e99c0b7fb68f6a6a69/vendor/github.com/aliyun/aliyun-oss-go-sdk/oss/conn.go#L772

Reply:
Done! move value check to a separate validate function is great! 
Thanks for your fast reply!